crates/deltalake-core/src/operations/cast.rs

```diff
@@ -37,6 +38,16 @@ fn cast_record_batch_columns(
         .collect::<Result<Vec<_>, _>>()
 }
 
+fn is_cast_required(a: &DataType, b: &DataType) -> bool {
+    match (a, b) {
+        (DataType::List(a_item), DataType::List(b_item)) => {
+            // If list item name is not the default('item') the list must be casted
+            !a.equals_datatype(b) || a_item.name() != b_item.name()
```

Comment:
I think just checking by name can give false positives, especially since we don't check the nested datatypes

Reply:
I also mean the other way around. This will return True if you only check by name:

        let field1 = DataType::List(FieldRef::from(Field::new("item", DataType::Int32, false)));
        let field2 = DataType::List(FieldRef::from(Field::new("element",  DataType::Float64,   false, )));

Reply:
One thing I dont follow, how come  !a.equals_datatype(b) not already return here True without a_item.name, Does it only check the top-level datatype with the name of the inner field?

Reply:
They only check the datatypes, but not the name.

`Compares the datatype with another, ignoring nested field names`
[It would be best to adjust this function, but it is native arrow](https://github.com/apache/arrow-rs/blob/master/arrow-schema/src/datatype.rs#L415)

Reply:
I see, then this fix should be fine since it does check nested types.